internal/net/http/transport/roundtrip_test.go

```diff
@@ -263,44 +135,165 @@ func Test_ert_RoundTrip(t *testing.T) {
+						return &http.Response{
+							Status: "200",
+						}, nil
+					},
+				},
+				bo: &backoffMock{
+					DoFunc: func(ctx context.Context, fn func() (interface{}, error)) (interface{}, error) {
+						return fn()
+					},
+				},
+			},
+			want: want{
+				wantRes: &http.Response{
+					Status: "200",
+				},
+			},
+		},
+		{
+			name: "return default roundtrip response if backoff is not nil",
+			args: args{
+				req: &http.Request{},
+			},
+			fields: fields{
+				transport: &roundTripMock{
+					RoundTripFunc: func(*http.Request) (*http.Response, error) {
+						return &http.Response{
+							Status: "200",
+						}, nil
+					},
+				},
+				bo: &backoffMock{
+					DoFunc: func(ctx context.Context, fn func() (interface{}, error)) (interface{}, error) {
+						return fn()
+					},
+				},
+			},
+			want: want{
+				wantRes: &http.Response{
+					Status: "200",
+				},
+			},
+		},
+		{
+			name: "return default roundtrip response if backoff use the default roundtrip",
+			args: args{
+				req: &http.Request{},
+			},
+			fields: fields{
+				transport: &roundTripMock{
+					RoundTripFunc: func(*http.Request) (*http.Response, error) {
+						return &http.Response{
+							Status: "200",
+						}, nil
+					},
+				},
+				bo: &backoffMock{
+					DoFunc: func(ctx context.Context, fn func() (interface{}, error)) (interface{}, error) {
+						return fn()
+					},
+				},
+			},
+			want: want{
+				wantRes: &http.Response{
+					Status: "200",
+				},
+			},
+		},
+		{
+			name: "return backoff error",
+			args: args{
+				req: &http.Request{},
+			},
+			fields: fields{
+				bo: &backoffMock{
+					DoFunc: func(ctx context.Context, fn func() (interface{}, error)) (interface{}, error) {
+						return nil, errors.New("error")
+					},
+				},
+			},
+			want: want{
+				err: errors.New("error"),
+			},
+		},
+		{
+			name: "return default roundtrip error if backoff use the default roundtrip",
+			args: args{
+				req: &http.Request{},
+			},
+			fields: fields{
+				transport: &roundTripMock{
+					RoundTripFunc: func(*http.Request) (*http.Response, error) {
+						return nil, errors.New("error")
+					},
+				},
+				bo: &backoffMock{
+					DoFunc: func(ctx context.Context, fn func() (interface{}, error)) (interface{}, error) {
+						return fn()
+					},
+				},
+			},
+			want: want{
+				err: errors.New("error"),
+			},
+		},
+		{
+			name: "return retryable error",
+			args: args{
+				req: &http.Request{},
+			},
+			fields: fields{
+				transport: &roundTripMock{
+					RoundTripFunc: func(*http.Request) (*http.Response, error) {
+						return nil, errors.Wrap(errors.ErrTransportRetryable, "error")
+					},
+				},
+				bo: &backoffMock{
+					DoFunc: func(ctx context.Context, fn func() (interface{}, error)) (interface{}, error) {
+						return fn()
+					},
+				},
+			},
+			want: want{
+				err: errors.Wrap(errors.ErrTransportRetryable, "error"),
+			},
+		},
 	}
 
+	log.Init()
```

Comment:
It panics depending on the order of the test cases that call log.Init. So could you please add it to TestMain.?

Reply:
Oh I didn't know that before. fixed.